proxywasm/internal/hostcall_utils.go

```diff
@@ -18,10 +18,17 @@ import (
 	"unsafe"
 )
 
+// StringBytePtr uses unsafe to convert a string to a raw byte pointer.
```

Comment:
This modification would prevent users who are already using higress from compiling with Go versions prior to 1.20. Can we differentiate this using Go build tags?
Retain the previous logic and ensure compatibility through go:build tinygo_slice_header
Add the corresponding tag here:https://github.com/alibaba/higress/blob/cfa3baddf8cf1978b954c7739f0a97ba7973888a/plugins/wasm-go/Dockerfile#L20

Reply:
I have updated this and changed description. Thanks @johnlanni!

Reply:
> Retain the previous logic and ensure compatibility through go:build tinygo_slice_header.

I have used go version in build tag to ensure compatibility, for example - https://github.com/mongodb/mongo-go-driver/blob/master/internal/errutil/join_go1.19.go

Reply:
Okay, to confirm with you, I understand that when compiling Go, it does not automatically include the version number as a tag, so here the user actually needs to explicitly specify --tags="go1.20", right?

Reply:
It is automatic @johnlanni. You can check https://pkg.go.dev/cmd/go#hdr-Build_constraints.

https://go.dev/doc/toolchain#config
https://stackoverflow.com/a/38439941

Reply:
Thanks!